pinot-spi/src/main/java/org/apache/pinot/spi/executor/HardLimitExecutor.java

```diff
@@ -38,12 +39,21 @@ public class HardLimitExecutor extends DecoratorExecutorService {
   private final AtomicInteger _running;
   private final int _max;
   private final QueryThreadExceedStrategy _exceedStrategy;
+  private final Consumer<Integer> _currentUsageGaugeUpdater;
 
-  public HardLimitExecutor(int max, ExecutorService executorService, QueryThreadExceedStrategy exceedStrategy) {
+  public HardLimitExecutor(int max, ExecutorService executorService, QueryThreadExceedStrategy exceedStrategy,
+      Consumer<Integer> maxUsageGaugeUpdater, Consumer<Integer> currentUsageGaugeUpdater) {
     super(executorService);
     _running = new AtomicInteger(0);
     _max = max;
     _exceedStrategy = exceedStrategy;
+    _currentUsageGaugeUpdater = currentUsageGaugeUpdater;
+    maxUsageGaugeUpdater.accept(max);
+    _currentUsageGaugeUpdater.accept(0);
+  }
+
+  public HardLimitExecutor(int max, ExecutorService executorService, QueryThreadExceedStrategy exceedStrategy) {
+    this(max, executorService, exceedStrategy, max1 -> { }, current -> { });
```

Comment:
(format) Seems this is not following pinot style. Same for other places

Reply:
It is not violating check style, but just not follow Pinot Style (follow [this](https://docs.pinot.apache.org/developers/developers-and-contributors/code-setup#set-up-ide) to set it up in your IDE)